orsopy/fileio/base.py

```diff
@@ -34,17 +37,113 @@ def __datetime_representer(dumper, data):
 yaml.add_representer(datetime.datetime, __datetime_representer)
 
 
-class Header:
+class HeaderMeta(type):
```

Comment:
Having gone through the code more closely I now understand how the Metaclass works to construct the _orso_optionals from the typing. However, I do wonder how robust this will be in the long run.